tools/topology/topology1/sof-acp-rmb-tdm8ch-dmic4ch-rt5682-rt1019.m4

```diff
@@ -33,13 +32,17 @@ VIRTUAL_WIDGET(ACPHS1 OUT, output, 3)
 #
 # Define the pipelines
 #
-# PCM0 ----> buffer ------+
-#                         |----> muxdemux ----> buffer ---->  ACPSP_play
-# PCM1 ----> buffer ----->
 #
-# PCM0 <---- muxdemux <---- buffer <---- ACPSP_Capture
+#              |-------------->echoref --->ACPHS1_Playback
+#              |
+# PCM3 ----> buffer ------+
```

Comment:
again very confused by linking at buffer

Reply:
I think it is the confusion with the name mentioned in the topology as it is recommended for reference from one of the Intel's topology which was used for the echo-ref. On AMD platform, it is intended to use for playback only. On AMD platform, we need a virtual/dummy DAI to support simultaneous playback on speaker and headset using single I2S controller instance, hence the connection of the buffer is to MUX and the reference buffer. So, connections to ACPHS1.OUT are kind of dummy only to get some of the required IPC messages from driver to enable two different PCM streams for headset and the speaker for simultaneous playback. It was the need of the design/framework, but improper naming confused.  We will update the patch with proper name and resubmit.  Let me know if any further queries.

Reply:
> We will update the patch with proper name and resubmit.

Some inline comments can't hurt either. When a relatively long explanation is needed on Github, some of it most likely belong to the corresponding source. It should not be required to use git or (even less convenient) github to understand what some source does.

My 2 cents.

Reply:
The echo-ref is a virtual component without any functionality (echo-ref will not read/write from/to buffer), it is used only for termination of the ACPHS1 (2nd playback-dai of the mux).